File: Quinto_Semestre/UVA/SuperSale/teste.py
fat = []

def fator(n):
    if n == 0:
        fat.append(0);
    elif n == 1:
        fat.append(1);
    else:
        fat.append(n*fat[n-1])

n = int(input())

i = 0
while i <= n:
    fator(i)
    i += 1

print(fat[n])

File: Quinto_Semestre/UVA/SuperSale/teste.cpp
#include <bits/stdc++.h>

using namespace std;

unsigned long long fat[1000001];

void fator(int n){
    if(n == 0){
        fat[n] = 0;
    }else if(n == 1){
        fat[n] = 1;
    }else if(n == 2){
        fat[n] = 2;
    }else{
        fat[n] = n * fat[n-1];
    }
}

int main (){

    int n;
    scanf("%d", &n);

    for(int i = 1; i < n+1; i++){
        fator(i);
    }

    printf("%llu\n", fat[n]);

    return 0;
}
